### DISCUSSION TITLE
Slidev/Vite: vite.config.ts Not Being Loaded

### DISCUSSION THREAD

**Discussion by zillionare** [slidevjs/slidev#1756] [Q&A]
When running my Slidev project, the vite.config.ts file does not seem to be loaded, resulting in custom configurations having no effect.

I've installed Slidev globally on my machine, and also installed several versions of Slidev in different locations, which might be the root of the problem. But I don't know how to go about checking this. If I run Slidev through the dev command specified in the package.json file within the project directory, then the vite.config.ts in the root directory gets loaded, and everything works as expected.

I am running npx slidev in the root directory of a project. In this directory, there is a package.json and a vite.config.ts. The vite.config.ts is correct; it's apparent that when I run npx slidev my-doc.md -t path/to/theme, it ultimately points somewhere else and thus fails to load the vite.config.ts; however, if I run it via npm run dev, the vite.config.ts in the root directory gets loaded, but I am unable to specify a theme.

Thanks in advance if anyone can help

**Comment by zillionare** [comment]
this almost drives me crazy but finally I found why.

when theme is specified, slidev will look for vite.config.ts from theme folder. In my use case, I have installed slidev in some working dir, say, ~/aaron/slidev, then eject some default theme and made customization. And all my slides are totally in a seperate folder which is backed by git.

so here is my usage:

```
cd ~/aaron/slidev
npx slidev /absolute/path/to/slides -t /absolute/path/to/theme
```

vite.config.ts at ~/aaron/slidev is never looked up for.

I switch themes a lot because I use slidev heavilly, so this maybe a typical use case. Maybe the behaviour could be documented somewhere.

Hope this will help others may confront same issue.

**Reply by kermanx** [reply]
How about specifying the theme in the frontmatter:

```md
---
theme: /absolute/path/to/theme
---
```
